samples/components/dialects/SQLDialect.tdd

```diff
@@ -0,0 +1,273 @@
+<?xml version='1.0' encoding='utf-8' ?>
+
+<!--
+  This dialect is the implementation of SQLDialect and BaseDialect.
```

Comment:
Does this implement BaseDialect? It looks like it's just for SQLdialect.

Reply:
yes, it implemented BaseDialect as well, actually only FormatSetIsolationLevel and FormatStartOfWeek are from basedialect.
Because there is not too much dialect implementation in basedialect, the majority of methods there are some dialect internal functions or helper functions to be used by individual dialects.

Reply:
SQLDialect is an abstract class, no datasource can use it directly, it has to be used as a base and subclass from it. I don't think it can pass tdvt test. I think the idea of this is to provide details of our base implementation of sqldialect/basedialect, so partners can understand it well while developing their own tdd.